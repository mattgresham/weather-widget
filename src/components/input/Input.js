// @flow
import React from 'react';
import styles from './Input.module.css';

type Props = {
  onChange: Function,
  placeholder?: string,
  value?: string
};

const Input = ({ onChange, placeholder, value }: Props) => (
  <input
    className={styles.element}
    onChange={onChange}
    placeholder={placeholder}
    type="text"
    value={value}
  />
);

Input.defaultProps = {
  placeholder: '',
  value: ''
};

export default Input;
