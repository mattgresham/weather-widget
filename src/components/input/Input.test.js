import React from 'react';
import { render } from 'enzyme';
import toJson from 'enzyme-to-json';
import Input from './Input';

const mockFn = jest.fn();

it('renders correctly', () => {
  const component = render(<Input onChange={mockFn} placeholder="Title" />);
  expect(toJson(component)).toMatchSnapshot();
});
