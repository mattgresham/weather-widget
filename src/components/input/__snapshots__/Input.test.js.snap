// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<input
  class="element"
  placeholder="Title"
  type="text"
  value=""
/>
`;
